### PR TITLE
Removed redundant calculations 

### PR DESCRIPTION
**Pull request by abhinavagarwal07**
Removed redundant calculations of checking imaginary equal zero as per #676

### CODE REVIEW

File: symengine/complex.h [hunk 1]
```diff
@@ -64,15 +64,15 @@ class Complex : public Number {
...
-    //! \return `true` if `-1`
-    virtual bool is_minus_one() const { return ((this->real_ == -1) and (this->imaginary_ == 0)); }
-    //! \return `true` if both `real_` or `imaginary_` are zero
+    //! \return false since imaginary cannot be zero
+    virtual bool is_zero() const { return false; }
+    //! \return false since imaginary cannot be zero
+    virtual bool is_one() const { return false; }
+    //! \return false since imaginary cannot be zero
+    virtual bool is_minus_one() const { return false; }
+    //! \return `true` if both `real_` is zero
```
Comment (certik) [inline symengine/complex.h]: Remove the word `both`. Perhaps you can say:
```
 //! \return `true` if `real_` is zero (since imaginary cannot be zero)
```

**Comment by certik** [comment]
Otherwise I think this is +1. @isuruf do you see any issues with this patch?

File: symengine/complex.h (hunk 1 above)
Comment (isuruf) [inline symengine/complex.h]: `is_reim_zero()` and `is_re_zero()` are both the same now. I would replace `is_reim_zero()` calls with `is_re_zero()`.

**Comment by abhinavagarwal07** [comment]
anything else needed ?

**Comment by isuruf** [comment]
Can you remove the method `is_reim_zero()` and change all calls to `is_re_zero()` as they are both the same now?

**Comment by abhinavagarwal07** [comment]
Their are no calls to ```is_reim_zero()```.

**Comment by isuruf** [comment]
Ah then, I'm +1 to merge after tests pass.

**Comment by Sumith1896** [comment]
+1 to merge!

**Comment by Sumith1896** [comment]
Thanks @abhinavagarwal07 :)

**Comment by abhinavagarwal07** [comment]
My pleasure

**Comment by certik** [comment]
Thanks @abhinavagarwal07 !

**Pull request merged**